File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Idexperience.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Idexperience implements ITableOperate {
 public final static String TAB_FILE_DATA = "/idexperience.txt";
 public static final int ID_ID_EXPERIENCE = 0 ;
public static final int ID_ID_PHYSICALVALUE = 1 ;
public static final int ID_ID_COST = 2 ;
public static final int ID_BAG_LIMIT = 3 ;
public static final int ID_FRIEND_LIMIT = 4 ;
public static final int MAX_RECORD = 5 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_BagLimit;
 public int GetBagLimit() { return m_BagLimit;}

private int m_FriendLimit;
 public int GetFriendLimit() { return m_FriendLimit;}

private int m_IDCost;
 public int GetIDCost() { return m_IDCost;}

private int m_IDExperience;
 public int GetIDExperience() { return m_IDExperience;}

private int m_IDPhysicalValue;
 public int GetIDPhysicalValue() { return m_IDPhysicalValue;}

public boolean LoadTable(HashMap<Integer, Table_Idexperience> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Idexperience> _hash = (HashMap<Integer, Table_Idexperience> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Idexperience _values = new Table_Idexperience();
 _values.m_BagLimit =  Integer.parseInt((String)valuesList.get(ID_BAG_LIMIT));
_values.m_FriendLimit =  Integer.parseInt((String)valuesList.get(ID_FRIEND_LIMIT));
_values.m_IDCost =  Integer.parseInt((String)valuesList.get(ID_ID_COST));
_values.m_IDExperience =  Integer.parseInt((String)valuesList.get(ID_ID_EXPERIENCE));
_values.m_IDPhysicalValue =  Integer.parseInt((String)valuesList.get(ID_ID_PHYSICALVALUE));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/tabletools/CodeTable/CSharp/Table_Idexperience.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Idexperience : ITableOperate
{ private const string TAB_FILE_DATA = "idexperience.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_ID_EXPERIENCE,
ID_ID_PHYSICALVALUE,
ID_ID_COST,
ID_BAG_LIMIT,
ID_FRIEND_LIMIT,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_BagLimit;
 public int BagLimit { get{ return m_BagLimit;}}

private int m_FriendLimit;
 public int FriendLimit { get{ return m_FriendLimit;}}

private int m_IDCost;
 public int IDCost { get{ return m_IDCost;}}

private int m_IDExperience;
 public int IDExperience { get{ return m_IDExperience;}}

private int m_IDPhysicalValue;
 public int IDPhysicalValue { get{ return m_IDPhysicalValue;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Idexperience _values = new Tab_Idexperience();
 _values.m_BagLimit =  Convert.ToInt32(valuesList[(int)_ID.ID_BAG_LIMIT] as string);
_values.m_FriendLimit =  Convert.ToInt32(valuesList[(int)_ID.ID_FRIEND_LIMIT] as string);
_values.m_IDCost =  Convert.ToInt32(valuesList[(int)_ID.ID_ID_COST] as string);
_values.m_IDExperience =  Convert.ToInt32(valuesList[(int)_ID.ID_ID_EXPERIENCE] as string);
_values.m_IDPhysicalValue =  Convert.ToInt32(valuesList[(int)_ID.ID_ID_PHYSICALVALUE] as string);

 _hash.Add(nKey,_values); }


}
}

